### PR TITLE
Theming

### PR DESCRIPTION
**Pull request by tof3d**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,3 +16,4 @@
 /bin/
 /composer.phar
 *~
+/nbproject/private/
```
Comment (AJahJahRiddim) [inline .gitignore]: remove this pls

File: .gitignore [hunk 2]
```diff
@@ -2,13 +2,13 @@
...
 /web/js/
 /web/uploads/
+/web/fonts/
 /app/bootstrap.php.cache
 /app/cache/*
 /app/config/parameters.yml
 /app/logs/*
 !app/cache/.gitkeep
 !app/logs/.gitkeep
-/app/data/*
```
Comment (AJahJahRiddim) [inline .gitignore]: Are u sure u wanna remove this folder ?? IDC but others might loose their db ...

File: app/config/config.yml [hunk 3]
```diff
@@ -50,30 +50,15 @@ assetic:
             inputs:
                 - %kernel.root_dir%/../vendor/twbs/bootstrap/dist/css/bootstrap.css
                 - %kernel.root_dir%/../vendor/twbs/bootstrap/dist/css/bootstrap-theme.css
-                - %kernel.root_dir%/../src/AppBundle/Resources/css/override.css
                 - %kernel.root_dir%/../vendor/flipclock/flipclock/compiled/flipclock.css
                 - %kernel.root_dir%/../src/AppBundle/Resources/css/flipclock.css
+                - %kernel.root_dir%/../src/AppBundle/Resources/css/spectrum.css
```
Comment (Garcia6l20) [inline app/config/config.yml:55]: well it seems you didn't add it !
git add src/AppBundle/Resources/css/spectrum.css

**Comment by AJahJahRiddim** [comment]
Im close to have a successful merge but from now spectrum.css is missing, pls add it.

**Comment by AJahJahRiddim** [comment]
It seems your specrum library comes from <https://github.com/bgrins/spectrum> right ?

If so you might use a vendor to include it to the project...
Take a look at [composer.json](https://github.com/ariescdi/documentio/blob/develop/composer.json) in repositories section.
Extra-doc [here](https://getcomposer.org/doc/05-repositories.md).

**Comment by AJahJahRiddim** [comment]
Successfully merged.
@tof3d  pls check the way I included the library in composer.json.

**Comment by valeriangalliat** [comment]
Thank you for the work.

A few points:

1. You merged a commit named "unstable commit to continue away ...", another one with an empty message, a merge commit from `develop` to `theming` branch and miscellaneous fixes on the *same feature branch*.

  Why this is bad on many levels? We work on features branches so we can merge them once they're fully done, and working. This means the *non-meaningful* progress commits (like "unstable commit to continue", empty commits, and fixes) can be *squashed* before merging so they don't pollute the history. Also there should be no merge commits on a feature branch since nobody cares you had to update your local working commit; you must *rebase* to update a feature branch.

1. The first commit contains the vendor library, which is wrong as highlighted by @AJahJahRiddim. The benefit of feature branches is that we can, again, rewrite the history to transparently fix these mistakes. There should be no trace in the history of the vendor being tracked (avoiding everyone to download useless vendor files in the history everytime they clone).

I insist that we do this again in good standing, so I'm reverting the changes made tonight temporarily. @tof3d, I urge you to push again your `theming` branch so we can work on it thursday.

**Note:** all the changes made tonight are backuped on the `develop-theming` branch.

**Comment by AJahJahRiddim** [comment]
Ok, as u wish, however I checked personally the new feature and It almost
functional, I checked the history before merging it and the 'to continue'
message was quite old ... I thought he tof made commits without good
description of what was actually made ...
Furthermore a merge request means 'job is done', doesn't it ?

I'll check you next time before taking the initiative by myself.

Cheers.

2015-03-08 9:48 GMT+01:00 Valérian Galliat <notifications@github.com>:

**Comment by valeriangalliat** [comment]
Yeah, it's not about the feature to be functional, it's about keeping the Git history clean.

Indeed, the "to continue" commit was old, and for sure have been continued in further commits, but the point is this should not appear in the main branch history. This is just temporary work, that, once completed, have to be rebased on a single, complete and working feature commit that can be merged. This is especially true for the merge commit from `develop` to the feature branch, which is completely meaningless in the general history.

A merge request often means the job is done (and it was effectively the case here), but even if the job is done, it's not always ready to merge (in this case, it's only that a rebase was seriously needed).

I could do the rebase right away, then merge the feature, but I don't think we need it right away, and I'd prefer we do this together on Thursday so we can make a practical course on rebasing and squashing commits.

> I'll check you next time before taking the initiative by myself.

No problem mate, Git allow us to make mistakes!

**Comment by AJahJahRiddim** [comment]
Noticed


2015-03-08 12:43 GMT+01:00 Valérian Galliat <notifications@github.com>:

**Comment by tof3d** [comment]
hey dudes,

@AJahJahRiddim 
Ok well I know how to add a vendor, I just didn't saw it, I asked myself if I create or not a repository for that and I finaly didn't do it coz that not mine, but to my defense, I dont put it directly on the web folder, I guess that's not so worst to make it copy from the AppBundle to the web folder.
So, I appreciate the intention to merge my work with the group one, but in spite of blame who was made to me I hold has to say the merge was not done as well also.
I saw you spent a lot of time for that...

@valeriangalliat 
Thanks for the smart and precise explanation, the good thing is that right now I know why I use an other branch.
I fixed my work and made a commit with something more global to explain waht I done and I sent a pull request, so can you push it to the develop branch and on you server to show something to arnault on thursday?

Cheers,

**Pull request merged**